modules/tinymce/src/core/main/ts/api/dom/DOMUtils.ts

```diff
@@ -1121,6 +1122,22 @@ const DOMUtils = (doc: Document, settings: Partial<DOMUtilsSettings> = {}): DOMU
     return state;
   };
 
+  const getContentEditableRoot = (node: Node) => {
```

Comment:
Where's the tests for this new public API? Also how is this any different to the above now as we are **not** returning the root anymore? Instead we're returning a true/false string based editable state if anything in the tree is a CEF element which is drastically different to what this used to do and cannot be used as a replacement for the old `NewLineAction` logic at the very least since it'll fail if you do something like this and pass in the `editable` text node (the editable root is the CET span in this case):

```html
<p contenteditable="false">Some <span contenteditable="true">editable</span> content</p>
```

Please have a discussion with either Morgan or Andy, or the wider team on how this public API should work, as it looks like it's just being thrown together without out any real consideration for API design sorry.

Reply:
The other solution was to have three different instances of the same function, in `NewLineUtils`, lists `Utils`, and Advlist `Utils` respectively. It was discussed that this would be better as a function in the `DOMUtils` API, however if you have a better solution I would appreciate it.

Reply:
Function has been removed from DOMUtils API and changes have been reverted.